packages/shared/src/components/modals/LockedSquadModal.tsx

```diff
@@ -31,14 +32,21 @@ const options: PromptOptions = {
 };
 
 type LockedSquadModalProps = {
-  squad: Squad;
+  squadId: string;
 } & ModalProps;
+
 function LockedSquadModal({
-  squad,
+  squadId,
   onRequestClose,
 }: LockedSquadModalProps): ReactElement {
+  const { data: squad, isLoading } = useQuery(
+    [{ type: 'squad_locked', squadId }],
+    ({ queryKey: [{ squadId: id }] }) => getSquad(id),
+  );
```

Comment:
Potentially you can set initialData to the squad you received actually.
Might make a slight difference in rendering for slow connections.

Let me know what you think.

Reply:
Great idea!